staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -472,80 +478,65 @@ type processorListener struct {
 
 func newProcessListener(handler ResourceEventHandler, requestedResyncPeriod, resyncPeriod time.Duration, now time.Time) *processorListener {
 	ret := &processorListener{
-		pendingNotifications:  []interface{}{},
 		nextCh:                make(chan interface{}),
+		addCh:                 make(chan interface{}),
```

Comment:
It isn't quite the same. The previous code grabbed a lock added it to a slice and unlocked. The new code must wait until you have something pull off the channel.
If you have a backlog of pending notifications, anything calling add will wait until pop actually gets to it.
I am a bit concerned that the Benchmark is just that and may not simulate reality of all the various ways add() is actually invoked.